src/Runner.Worker/Handlers/ScriptHandler.cs

```diff
@@ -321,12 +321,19 @@ public async Task RunAsync(ActionRunStage stage)
             ExecutionContext.Debug($"{fileName} {arguments}");
 
             Inputs.TryGetValue("standardInInput", out var standardInInput);
-            using (var stdoutManager = new OutputManager(ExecutionContext, ActionCommandManager))
-            using (var stderrManager = new OutputManager(ExecutionContext, ActionCommandManager))
+            using (ScriptStalledManager stalledManager = new ScriptStalledManager(ExecutionContext, TimeSpan.FromMinutes(10).TotalMilliseconds))
```

Comment:
nit: IMO double this to 20 or 30. Jobs timeout by default after 6 hours for hosted machines, we should optimize to limit false positives, not responsiveness for users as most users will just see the annotations/build logs after the fact and won't be monitoring in real time.

Reply:
Upped it to 30 minutes in [cda3bfc](https://github.com/actions/runner/pull/2093/commits/cda3bfcd90cdd1863e726e2fc23b94542da36a56) :)